### PR TITLE
Fix: Switch’s Usage of ImGui

### PR DESCRIPTION
**Pull request by dcvz**
Fixes up the Switch platform with some lost changes

### CODE REVIEW

File: src/port/switch/SwitchImpl.cpp [hunk 1]
```diff
@@ -74,6 +76,51 @@ void Ship::Switch::SetupFont(ImFontAtlas* fonts) {
     plExit();
 }
 
+void Ship::Switch::ImGuiSwapABXY(int startEvent) {
+    if (CVarGetInteger("gOpenMenuBar", 0)) {
+        return;
+    }
+
+    auto& input_queue = ImGui::GetCurrentContext()->InputEventsQueue;
+    for (int n = start_event; n < input_queue.Size; n++) {
```
Comment (briaguya0) [inline src/port/switch/SwitchImpl.cpp]:
```suggestion
    for (int n = startEvent; n < input_queue.Size; n++) {
```

to fix a build error

File: src/port/switch/SwitchImpl.cpp [hunk 2]
```diff
@@ -74,6 +76,51 @@ void Ship::Switch::SetupFont(ImFontAtlas* fonts) {
     plExit();
 }
 
+void Ship::Switch::ImGuiSwapABXY(int startEvent) {
+    if (CVarGetInteger("gOpenMenuBar", 0)) {
+        return;
+    }
```
Comment (briaguya0) [inline src/port/switch/SwitchImpl.cpp]: this part makes no sense to me. in my testing, it makes it so we aren't doing the button switching at all when using the imgui menus. This means we have the switch "B" button acting as "confirm" and the switch "A" button acting as "cancel"

Comment (briaguya0) [reply src/port/switch/SwitchImpl.cpp]: to add to this, I tried commenting this block out and I'm running into the issue where I can only press a button once when navigating the menu before input stops working and the controller can no longer be used to navigate the menu

Comment (dcvz) [reply src/port/switch/SwitchImpl.cpp]: Yeah, this was a misunderstanding. I thought we don’t actually want this in menus only in game

**Review by briaguya0 — APPROVED** [review]
Tested here https://github.com/HarbourMasters/Shipwright/pull/2445/commits/1fd3838d8145b82fc31fd29b2b0258f288249c08

I can confirm this works as expected

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**